src/components/stateful-enroll/StatefulEnroll.jsx

```diff
@@ -61,60 +56,32 @@ const StatefulEnroll = ({
     ...labels,
   };
 
-  const handleRedemptionError = () => {
-    setEnrollButtonState('error');
-    if (onError) {
-      onError();
-    }
-  };
-
-  const redemptionMutation = useRedemptionMutation({
-    onMutate: () => {
-      setEnrollButtonState('pending');
-    },
-    onSuccess: (transaction) => {
-      setTransactionStatusApiUrl(transaction.transactionStatusApiUrl);
-    },
-    onError: () => {
-      handleRedemptionError();
-    },
-  });
-
-  useTransactionStatus({
+  const { redeem } = useStatefullEnroll({
     contentKey,
-    transactionStatusApiUrl,
+    subsidyAccessPolicy,
+    onRedeem: () => {
+      setButtonState('pending');
+    },
     onSuccess: (transaction) => {
-      if (transaction.state === 'committed') {
-        setEnrollButtonState('complete');
-        if (onSuccess) {
-          onSuccess(transaction);
-        }
-      }
-      if (transaction.state === 'failed') {
-        handleRedemptionError();
+      setButtonState('complete');
+      if (onSuccess) {
+        onSuccess(transaction);
       }
     },
-    onError: () => {
-      handleRedemptionError();
-    },
   });
 
-  const handleEnrollButtonClick = (e) => {
+  const handleEnrollButtonClick = () => {
     if (onClick) {
-      onClick(e);
+      onClick();
     }
-    redemptionMutation.mutate({
-      userId: getAuthenticatedUser().id,
-      contentKey,
-      policyRedemptionUrl: subsidyAccessPolicy.policyRedemptionUrl,
-    });
+    redeem();
```

Comment:
This is probably where the try catch would need to be if you wanted it to fail more gracefully before setting the state of the `StatefulButton` to `'error'`

Reply:
@brobro10000 Ah, fwiw this code path wasn't being called related to our earlier conversation this morning. The error we were discussing was being thrown in `UserEnrollmentForm`'s call to `redeem`.

Reply:
[update/inform] Added an additional `try/catch` elsewhere to fix the issue and fail more gracefully.